Fixing analysis for file only

eddynaka opened the pull request:
Fixes #42

michaelcfanning commented on Src/SarifPatternMatcher/SearchSkimmer.cs:
```diff
@@ -175,98 +175,150 @@ public override void Analyze(AnalyzeContext context)
+                matchExpression.Level :
+                DefaultConfiguration.Level;
+
+            if (!string.IsNullOrEmpty(matchExpression.ContentsRegex))
```
There is significant conditional statement complexity in this method, we should start to refactor into methods,

eddynaka replied:
sure! separated in two methods.

---
In reply to: [545985868](https://github.com/microsoft/sarif-pattern-matcher/pull/44#discussion_r545985868) [](ancestors = 545985868)

michaelcfanning commented:
Great change! But I think we need to add some more focused unit tests that cover all possibilities in this scenario, it's too easy to miss changes that fly by in the log file. I checked in this code without observing that fake.sarif was so enormous for example! oops.

See SearchSkimmerTests for the beginnings of some patterns. Also consider whether we need a test at the AnalyzeCommand level, i'm not sure we do but interested in your opinion.

eddynaka commented:
just added more tests and fixed the code coverage. Let me know what do you think :)

---
In reply to: [748224385](https://github.com/microsoft/sarif-pattern-matcher/pull/44#issuecomment-748224385) [](ancestors = 748224385)

michaelcfanning commented on Src/Test.UnitTests.SarifPatternMatcher/SearchSkimmerTests.cs:
```diff
@@ -15,19 +15,35 @@ namespace Microsoft.CodeAnalysis.SarifPatternMatcher.SarifPatternMatcher
+        }
+
         [Fact]
-        public void SearchSkimmer_DetectsBase64EncodedPattern()
```
>SearchSkimmer_DetectsBase64EncodedPattern [](start = 20, length = 41)

This is our test naming convention to put the name of the type under test as a prefix. it helps with search in the test explorer. You probably don't like it because of style rules that flag the underscore?? :) we should have a stylecop ruleset specifically for test code that turns this one off.

eddynaka replied:
I just think that that is strange. Today, it's organized like this:
1. project name
  - namespace
     - class name
         - test name

So, in theory, its a duplicated work, since test name = classname_testname. Either way, both works :)

---
In reply to: [546093717](https://github.com/microsoft/sarif-pattern-matcher/pull/44#discussion_r546093717) [](ancestors = 546093717)

michaelcfanning reviewed (approved):
:shipit:
